src/qt/managenamespage.cpp

```diff
@@ -0,0 +1,289 @@
+          QMessageBox::Cancel))
+    {
+        return;
+    }
+
+    WalletModel::UnlockContext ctx(walletModel->requestUnlock());
+    if (!ctx.isValid())
+        return;
+
+    QString err_msg;
+    std::string strName = name.toStdString();
+
+    try
+    {
+        NameNewReturn res = walletModel->nameNew(name);
+
+        if (res.ok)
+        {
+            // save pending name firstupdate data ... this gets
+            // picked up after the config name dialog is accepted
+            pendingNameFirstUpdate[strName] = res;
+
+            // reset UI text
+            ui->registerName->setText("d/");
+            ui->submitNameButton->setDefault(true);
+
+            int newRowIndex;
+            // FIXME: CT_NEW may have been sent from nameNew (via transaction).
+            // Currently updateEntry is modified so it does not complain
+            model->updateEntry(name, "", NameTableEntry::NAME_NEW, CT_NEW, &newRowIndex);
+            ui->tableView->selectRow(newRowIndex);
+            ui->tableView->setFocus();
+
+            ConfigureNameDialog dlg(platformStyle, name, "", true, this);
+            dlg.setModel(walletModel);
+
+            if (dlg.exec() == QDialog::Accepted)
+            {
+                LOCK(cs_main);
+                if (pendingNameFirstUpdate.count(strName) != 0)
+                {
+                    model->updateEntry(name, dlg.getReturnData(), NameTableEntry::NAME_NEW, CT_UPDATED);
+                }
+                else
+                {
+                    // name_firstupdate could have been sent, while the user was editing the value
+                    // Do nothing
+                }
+            }
+
+            return;
+        }
+
+        err_msg = QString(res.err_msg.c_str());
+    }
+    catch (std::exception& e)
+    {
+        err_msg = e.what();
+        LogPrintf("ManageNamesPage::on_submitNameButton_clicked; %s", err_msg.toStdString().c_str());
+    }
+
+    if (err_msg == "ABORTED")
+        return;
+
+    QMessageBox::warning(this, tr("Name registration failed"), err_msg);
+}
+
+bool ManageNamesPage::eventFilter(QObject *object, QEvent *event)
+{
+    if (event->type() == QEvent::FocusIn)
+    {
+        if (object == ui->registerName)
+        {
+            ui->submitNameButton->setDefault(true);
+            ui->configureNameButton->setDefault(false);
+        }
+        else if (object == ui->tableView)
+        {
+            ui->submitNameButton->setDefault(false);
+            ui->configureNameButton->setDefault(true);
+        }
+    }
+    return QDialog::eventFilter(object, event);
+}
+
+void ManageNamesPage::selectionChanged()
+{
+    // Set button states based on selected tab and selection
+    QTableView *table = ui->tableView;
+    if(!table->selectionModel())
+        return;
+
+    if(table->selectionModel()->hasSelection())
+    {
+        ui->configureNameButton->setEnabled(true);
+    }
+    else
+    {
+        ui->configureNameButton->setEnabled(false);
+    }
```

Comment:
Why not `ui->configureNameButton->setEnabled(table->selectionModel()->hasSelection());`?